specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2021-12-01-preview/examples/ApiManagementListPolicyFragmentReferences.json

```diff
@@ -0,0 +1,24 @@
+{
+  "parameters": {
+    "serviceName": "apimService1",
+    "resourceGroupName": "rg1",
+    "api-version": "2021-12-01-preview",
+    "subscriptionId": "subid",
+    "policyFragmentId": "policyFragment1"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
+          {
+            "id": "/subscriptions/subid/resourceGroups/rg1/providers/Microsoft.ApiManagement/service/apimService1/policies/policy",
```

Comment:
I see. The API path '../Microsoft.ApiManagement/service/{serviceName}/policyFragments/{id}/references' is for listing references resources under the parent policyFragments resource. The type and id in the response should conform to that.

Can you change it to POST api with something like '/Microsoft.ApiManagement/service/{serviceName}/policyFragments/{id}/listReferences'?

Reply:
Why POST for a call that just returns a collection? We're not posting any payload nor invoking an action, just getting a collection.

`is for listing references resources under the parent policyFragments resource`
I wouldn't really say that there's a parent/child relationship between those resources. A policy fragment resource can be used by other resources. This API lists those other resources that use a policy fragment. But all of them can exist independently.
E.g. you create a policy fragment called fragment1. Then someone updates other resources(e.g. policy1 and policy2) and includes the policy fragment in them. Calling policyFragments/fragment1/references should then list policy1 and policy2, because they use this policy fragment.

Reply:
Listing is a valid POST action. (e.g. listKeys).
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md

Your intent is not a parent/child relationship, but GET on this api path indicates listing all child resources of type `Microsoft.ApiManagement/service/policyFragments/references` and the api should return resources of that type not others since it requires `Microsoft.ApiManagement/service/policyFragments/references/read` RBAC permission.

Having a POST action like `/Microsoft.ApiManagement/service/{serviceName}/policyFragments/{id}/listReferences` or
`/Microsoft.ApiManagement/listPolicyReferences` will avoid the issue.

Reply:
There are some things I might have explained wrongly or we didn't understand each other:
1. The resources are a child resource of the policy fragment. The type of the resources is the `common-types/resource-management/v2/types.json#/definitions/Resource`. So we need to reply with a collection of the common-type Resource resources that reference the policy fragment. See the `./definitions.json#/definitions/ResourceCollection` for what the definition of this response.
2. The common-type Resource contains Id, Name and Type.  We need those fields to represent the actual resources that reference the policy fragment. Check ApiManagementListPolicyFragmentReferences.json to see an example.
3. We also need pagination (which is currently missing in this PR, I need to add it). If this is a POST request, how do we pass the pagination top and skip parameters? I imagine we shouldn't have a POST request with params in the query string.

Given that we need to list child resources of the type `common-types/resource-management/v2/types.json#/definitions/Resource`, do you still recommend that we make this endpoint a POST?

Reply:
`service/policyFragments/referernces` and `service/policies` are all proxy resources, but it does not mean that the list API for one type can return a collection of the other type. It's a RPC/RBAC violation. The API `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ApiManagement/service/{serviceName}/policyFragments/{id}/references` is for listing `service/policyFragments/referernces` resources.

POST request can have a query string and support pagination.

Reply:
I've updated the endpoint to be POST and have pagination query string parameters.